sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/TokenManager.java

```diff
@@ -0,0 +1,60 @@
+package com.microsoft.appcenter.storage;
+
+import com.microsoft.appcenter.utils.storage.SharedPreferencesManager;
+import com.google.gson.Gson;
+import com.microsoft.appcenter.storage.models.TokenResult;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Set;
+
+public class TokenManager {
+    private final Gson gson; // TODO use Gson object from shared module.
+
+    private static TokenManager tInstance;
+
+    private TokenManager() {
+        gson = new Gson();
+    }
+
+    public static synchronized TokenManager getInstance() {
+        if (tInstance == null) {
+            tInstance = new TokenManager();
+        }
+        return tInstance;
+    }
+
+    public Set<String> getPartitionNamesSet() {
+        Set<String> partitionNamesSet = SharedPreferencesManager.getStringSet(Constants.PARTITION_NAMES);
+        return partitionNamesSet == null ? new HashSet<String>() : partitionNamesSet;
+    }
+
+    public TokenResult getCachedToken(String partitionName) {
+        TokenResult token = gson.fromJson(SharedPreferencesManager.getString(partitionName), TokenResult.class);
+        if (token != null){
+            long now = new Date().getTime();

```

Comment:
>getTime [](start = 34, length = 7)

does this return time in UTC? I think token.ttl was documented as UTC, right?

Reply:
ah yup! this is local time :see_no_evil: I used instant.now() before the change... but it's in Java 8... will check another way...